msu/hooks/skills/skill.nut

```diff
@@ -0,0 +1,1055 @@
+::mods_hookDescendants("skills/skill", function(o) {
+	if ("create" in o)
+	{
+		local create = o.create;
+		o.create = function()
+		{
+			create();
+			if (this.m.InjuriesOnBody != null)
+			{
+				this.setupDamageType();
+			}
+		}
+	}
+});
+
+::mods_hookBaseClass("skills/skill", function(o) {
+	o = o[o.SuperName];
+
+	o.m.DamageType <- [];
+	o.m.ItemActionOrder <- ::Const.ItemActionOrder.Any;
+
+	o.m.IsBaseValuesSaved <- false;
+	o.m.ScheduledChanges <- [];
+
+	o.scheduleChange <- function( _field, _change, _set = false )
+	{
+		this.m.ScheduledChanges.push({Field = _field, Change = _change, Set = _set});
+	}
+
+	o.executeScheduledChanges <- function()
+	{
+		if (this.m.ScheduledChanges.len() == 0)
+		{
+			return;
+		}
+
+		foreach (c in this.m.ScheduledChanges)
+		{
+			switch (typeof c.Change)
+			{
+				case "integer":
+					if (c.Set)
+					{
+						this.m[c.Field] = c.Change;
+					}
+					else
+					{
+						this.m[c.Field] += c.Change;
+					}
+					break;
+
+				case "string":
+					if (c.Set)
+					{
+						this.m[c.Field] = c.Change;
+					}
+					else
+					{
+						this.m[c.Field] += c.Change;
+					}
+					break;
+
+				default:
+					this.m[c.Field] = c.Change;
+					break;
+			}
+		}
+
+		this.m.ScheduledChanges.clear();
+	}
+
+	o.saveBaseValues <- function()
+	{
+		if (!this.m.IsBaseValuesSaved)
+		{
+			this.b <- clone this.skill.m;
+			foreach (k, v in this.m)
+			{
+				this.b[k] <- v;
+			}
```

Comment:
I'd be wary of that in case the `m` table stores references to other BB classes or tables. We wouldn't want to `deepClone` them. I think the current system is ok. We can put in the documentation that reference types are stored as references in b.

Reply:
Sounds good to me